eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -363,15 +367,37 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
    */
 
   when(NORMAL)(handleExceptions {
-    case Event(c: ForbiddenCommandDuringSplice, d: DATA_NORMAL) if d.spliceStatus != SpliceStatus.NoSplice =>
+    case Event(c: ForbiddenCommandDuringQuiescence, d: DATA_NORMAL) if d.spliceStatus.isInstanceOf[QuiescenceNegotiation] =>
+      val error = ForbiddenDuringQuiescence(d.channelId, c.getClass.getSimpleName)
+      c match {
+        case c: CMD_ADD_HTLC => handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
+        case _: HtlcSettlementCommand => stay() // htlc settlement commands will be ignored and replayed when not quiescent
+        case _ => handleCommandError(error, c)
+      }
+
+    case Event(c: ForbiddenCommandDuringSplice, d: DATA_NORMAL) if d.spliceStatus != SpliceStatus.NoSplice  && !d.spliceStatus.isInstanceOf[QuiescenceNegotiation] =>
       val error = ForbiddenDuringSplice(d.channelId, c.getClass.getSimpleName)
       c match {
         case c: CMD_ADD_HTLC => handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
-        // NB: the command cannot be an htlc settlement (fail/fulfill), because if we are splicing it means the channel is idle and has no htlcs
+        // NB: the command cannot be an htlc settlement (fail/fulfill), because if we are splicing without quiescence support it means the channel is idle and has no htlcs
```

Comment:
Actually at that point we can perfectly well be using quiescence (and have already reached the quiescent state), so the comment is incorrect. But more fundamentally, I think we should include the splicing protocol inside the quiescence timeout timer: we don't want to stay quiescent too long, because it creates a risk of pending HTLCs timing out and it makes the channel unable to forward HTLCs (and thus loose fee revenue). If either peer has a bug that makes the splicing protocol block, we want to disconnect to restore the channel to a known good state.

I've implemented this in https://github.com/ACINQ/eclair/commit/2476fb83a3440891fc6b24fbd76bcdb6e82b8c71

Reply:
Good idea to extend the quiescence time out period to include the splice negotiation. Fixed in [5841138](https://github.com/ACINQ/eclair/pull/2680/commits/58411385f7da40e50f5180a7185344317b608b73).